src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/set_interior_vehicle_data_request.cc

```diff
@@ -148,168 +170,184 @@ const std::string LightName(const smart_objects::SmartObject& light_name) {
-    if (message_params::kStatus == it->first ||
-        message_params::kId == it->first) {
+  std::map<std::string, std::string> mapping =
+      GetLightCapabilitiesMapping();
+
+  for (auto it = control_data.map_begin(); it != control_data.map_end(); ++it) {
+    const std::string& request_parameter = it->first;
+
+    if (message_params::kId == request_parameter) {
       continue;
     }
-    const std::string& caps_key = lightCapsMapping[it->first];
-    LOG4CXX_DEBUG(logger_,
-                  "Checking request parameter "
-                      << it->first << " with capabilities. Appropriate key is "
-                      << caps_key);
-
-    if (!capabilities.keyExists(caps_key)) {
-      LOG4CXX_DEBUG(logger_,
-                    "Capability "
-                        << caps_key
-                        << " is missed in RemoteControl capabilities");
-      return false;
-    }
-    if (!capabilities[caps_key].asBool()) {
-      LOG4CXX_DEBUG(logger_,
-                    "Capability "
-                        << caps_key
-                        << " is switched off in RemoteControl capabilities");
-      return false;
+
+    const mobile_apis::Result::eType item_capability =
+        GetItemCapability(capabilities,
+                          mapping,
+                          request_parameter,
+                          mobile_apis::Result::READ_ONLY);
+
+    if (mobile_apis::Result::SUCCESS != item_capability) {
+      return item_capability;
     }
   }
-  return true;
+
+  return mobile_apis::Result::SUCCESS;
 }
 
-bool CheckLightNameByCapabilities(
+mobile_apis::Result::eType GetLightNameCapabilities(
     const smart_objects::SmartObject& capabilities_status,
     const smart_objects::SmartObject& light_data) {
   LOG4CXX_AUTO_TRACE(logger_);
   auto it = capabilities_status.asArray()->begin();
+
   for (; it != capabilities_status.asArray()->end(); ++it) {
     const smart_objects::SmartObject& so = *it;
     const int64_t current_id = so[message_params::kName].asInt();
+    
     if (current_id == light_data[message_params::kId].asInt()) {
-      return CheckLightDataByCapabilities(so, light_data);
+      return GetLightDataCapabilities(so, light_data);
     }
   }
+
   LOG4CXX_DEBUG(logger_, "There is no such light name in capabilities");
-  return false;
+  return mobile_apis::Result::UNSUPPORTED_RESOURCE;
 }
 
-bool CheckControlDataByCapabilities(
-    const smart_objects::SmartObject& module_caps,
+module_capability GetControlDataCapabilities(
+    const smart_objects::SmartObject& capabilities,
     const smart_objects::SmartObject& control_data) {
   LOG4CXX_AUTO_TRACE(logger_);
   std::map<std::string, std::string> mapping =
       GetModuleDataToCapabilitiesMapping();
-  const smart_objects::SmartObject& capabilities_status = module_caps[0];
-  auto it = control_data.map_begin();
-  for (; it != control_data.map_end(); ++it) {
+
+  for (auto it = control_data.map_begin(); it != control_data.map_end(); ++it) {
```

Comment:
@KVGrygoriev maybe loop range style
for (const auto control_element : control_data) {

Reply:
@EKuliiev, It won't work because of SmartObject

Reply:
yes, my bad.